pages/kids/image-quiz/index.tsx

```diff
@@ -0,0 +1,162 @@
+    e.preventDefault();
+    e.stopPropagation();
+
+    if (correct) {
+      const textGood = text.good[getRandomIndex(text.good.length)];
+      AudioPlayer.getInstance()
+        .playSrc(matchSound)
+        .then(
+          async () =>
+            await AudioPlayer.getInstance().playTextToSpeech(
+              new Phrase_deprecated(textGood).getTranslation(currentLanguage),
+              currentLanguage
+            )
+        );
+      setTimeout(() => {
+        setPhrases(shuffle(phrases));
+        setCorrectIndex(getRandomIndex());
+      }, 4000);
+    } else {
+      const textWrong = text.wrong[getRandomIndex(text.wrong.length)];
+      AudioPlayer.getInstance()
+        .playSrc(dontMatchSound)
+        .then(
+          async () =>
+            await AudioPlayer.getInstance().playTextToSpeech(
+              new Phrase_deprecated(textWrong).getTranslation(currentLanguage),
+              currentLanguage
+            )
+        );
+    }
+  };
+
+  if (!phrases?.[correctIndex]) {
+    return null;
+  }
+
+  const otherTranslation = phrases[correctIndex].getTranslation(otherLanguage);
+
+  return (
+    <div className="bg-gradient-to-r from-[#fdf6d2] to-[#99bde4] -mb-8 -m-2">
+      <SEO
+        title={t(`seo.kids_page_imagequiz_title.${getCountryVariant()}`)}
+        description={t(`seo.kids_page_imagequiz_description.${getCountryVariant()}`)}
+        image="https://www.movapp.cz/icons/movapp-cover-kids.jpg"
+      />
+      <article className="flex flex-col m-auto items-center py-2 sm:py-4">
+        <header>
+          <KidsTranslation
+            language={otherLanguage}
+            transcription={phrases[correctIndex].getTranscription(otherLanguage)}
+            translation={otherTranslation}
+            soundUrl={phrases[correctIndex].getSoundUrl(otherLanguage)}
+          />
+        </header>
+        <div className="flex flex-wrap justify-center px-2 sm:px-4">
+          {phrases.slice(0, CHOICES_COUNT).map((phrase, index) => {
+            return (
+              <ImageContainer key={phrase.getTranslation('uk')} phrase={phrase} onClick={handleClick} correct={index === correctIndex} />
+            );
+          })}
+        </div>
+      </article>
+    </div>
+  );
+};
+
+const ImageContainer = ({ phrase, onClick, correct }: ImageContainerProps): JSX.Element => {
+  const { otherLanguage } = useLanguage();
+  const [className, setClassName] = useState('');
+
+  return (
+    <div
+      className={`max-w-sm rounded-2xl overflow-hidden shadow-xl w-72 m-5 md:m-8 bg-[#f7e06a] max-h-[34rem] ${className}`}
+      onClick={(e) => {
+        setClassName(correct ? styles.match : styles.dontMatch);
+        onClick(e, phrase, correct);
+      }}
+    >
+      <button className="w-72 h-72 relative bg-white">
+        <Image src={phrase.getImageUrl() ?? ''} layout="fill" sizes="100%" objectFit="cover" alt={phrase.getTranslation(otherLanguage)} />
```

Comment:
Improvmen idea (we can do it as separate ticket): 

Right now we load images when the user guesses correctly and need to wait for them. Instead, whenever we display new images we should also preload the next 3, so they are displayed instantly in the next round. 

You can preload images something like this: 

``` 
function preloadImage(url)
{
    var img=new Image();
    img.src=url;
}
```

and keep track of them using `useRef` :)

Reply:
I can add useRef array for preloaded images.

Reply:
Should I use `priority` property for `Image`?

Reply:
`priority` property does not really help here, we want to fetch the image before we actually render it. Don't worry, we can improve this later, I am myself not sure how exactly to achieve what I want without just testing it :) 

I think you can remove the `priority` for now.

Reply:
Fine, I removed it.